src/subRecordPicker.js

```diff
@@ -29,11 +35,12 @@ export function createSubrecordPicker(sruUrl, retrieveAll = false) {
     return new Promise((resolve, reject) => {
       const records = [];
       sruClient.searchRetrieve(`melinda.partsofhost=${recordId}`)
-        .on('record', xmlString => {
-          records.push(MARCXML.from(xmlString)); // eslint-disable-line functional/immutable-data
+        .on('record', async xmlString => {
+          records.push(await MARCXML.from(xmlString, {subfieldValues: false})); // eslint-disable-line functional/immutable-data
         })
-        .on('end', () => {
-          resolve(records);
+        .on('end', async () => {
+          await Promise.all(records);
+          resolve({records});
         })
```

Comment:
You need to handle possible async errors from`MARCXML.from` (Wrap the `end` callback in try catch and reject in case of error. Example: https://github.com/NatLibFi/melinda-record-harvest-harvester/blob/f685d57f43037dd6901de0c5e45a2320a4cfba67/src/app.js#L102)